src/bosh_openstack_cpi/spec/integration/lifecycle_spec.rb

```diff
@@ -243,6 +243,99 @@ def where_ip_address_is(ip)
     end
   end
 
+  context 'when booting from volume with a boot_volume_type' do
+    let(:boot_from_volume) { true }
+    let(:boot_volume_type) { @boot_volume_type }
+
+    let(:network_spec) do
+      {
+        'default' => {
+          'type' => 'manual',
+          'ip' => @manual_ip,
+          'cloud_properties' => {
+            'net_id' => @net_id
+          }
+        }
+      }
+    end
+
+    it 'exercises the vm lifecycle' do
+      expect {
+        vm_lifecycle(@stemcell_id, network_spec, [])
+      }.to_not raise_error
+    end
+  end
+
+  context 'when booting from volume with ephemeral disk and root disk size = 0' do
+    let(:boot_from_volume) { true }
+
+    let(:network_spec) do
+      {
+        'default' => {
+          'type' => 'manual',
+          'ip' => @manual_ip,
+          'cloud_properties' => {
+            'net_id' => @net_id
+          }
+        }
+      }
+    end
+
+    let(:resource_pool) do
+      {
+        'ephemeral_disk' => {
+          'size' => 0,
+          'type' => 'gp2'
+        },
+        'root_disk' => {
+          'size' => 0,
+          'type' => 'standard'
+        }
+      }
+    end
+
+    it 'exercises the vm lifecycle' do
+      expect {
+        vm_lifecycle(@stemcell_id, network_spec, [], resource_pool)
+      }.to_not raise_error
+    end
+  end
+
+  context 'when booting from volume with ephemeral disk and root disk size > 0' do
+    let(:boot_from_volume) { true }
+
+    let(:network_spec) do
+      {
+        'default' => {
+          'type' => 'manual',
+          'ip' => @manual_ip,
+          'cloud_properties' => {
+            'net_id' => @net_id
+          }
+        }
+      }
+    end
+
+    let(:resource_pool) do
+      {
+        'ephemeral_disk' => {
+          'size' => 3000,
+          'type' => 'gp2'
+        },
```

Comment:
see above, please remove the `ephemeral_disk` part and `type` definition in the `root_disk`